src/engine.cairo

```diff
@@ -168,8 +168,48 @@ fn bits_to_target(bits: u32) -> u256 {
     0
 }
 
-fn target_to_bits(target: u256) -> u32 {
-    0
+pub fn target_to_bits(target: u256) -> Result<u32, felt252> {
```

Comment:
It will be much easier to read if it would be implemented in terms of shl and shr operations, like it is implemented in ZeroSync repo:
https://github.com/ZeroSync/ZeroSync/blob/f1bd4736b0bac190ac48cbc8d5fcd92e2e1be7b9/src/block/block_header.cairo#L451
 


You can find shl and shr here: 
* https://github.com/keep-starknet-strange/joyboy/blob/4d747257b65cb46b6506bc65b18bbe3adff95658/onchain/src/utils.cairo#L72
* https://github.com/keep-starknet-strange/joyboy/blob/4d747257b65cb46b6506bc65b18bbe3adff95658/onchain/src/utils.cairo#L93

Reply:
Oh! I thought there were no left/right shift operators in Cairo! Thanks